src/server/hotkey_collector.cpp

```diff
@@ -65,7 +73,11 @@ void hotkey_collector::capture_hash_key(const dsn::blob &hash_key, int64_t weigh
     _internal_collector->capture_data(hash_key, weight);
 }
 
-void hotkey_collector::analyse_data() { _internal_collector->analyse_data(); }
+void hotkey_collector::analyse_data()
+{
+    terminate_by_timeout();
+    _internal_collector->analyse_data();
```

Comment:
How to comprehend the logic when collector is terminated but still execute analyse_data? I think you can add a boolean return value for `terminate_by_timeout `, and judge it at first.